### ISSUE TITLE
Error when using automock for variadic functions

### ISSUE THREAD

**Issue by Enes1313** [asomers/mockall#501]
Hello,
I'm encountering an error when using automock for variadic functions in the Mockall. The relevant code snippet is as follows:

![image](https://github.com/asomers/mockall/assets/12311762/3af52c34-b9da-4205-aa32-1cd8c9f303c7)

How can I resolve this error?

**Comment by asomers** [comment]
lol, the popup error is covering up the function that you're trying to mock.  Better just to write your question in plaintext rather than use an image.

But the answer is that you can't.  Mockall can't mock variadic functions because Rust doesn't have variadic functions.  The best that you can do will probably be to manually write a variadic macro that turns its arguments into a `Vec` and then passes them to a automock'd function.

**Comment by Enes1313** [comment]
Thank you for the answer. My main goal is a hierarchy and order to prepare unit and integration tests in Rust language for C projects. That's why bindgen and mockall projects are very important for me :). I wonder if this feature can be added even if it is a nightly feature? I can try to implement it too, if you're going to enable such a property. Apart from that, if you anticipate that there will be other big problems in my goal, I can listen to them as well from you.

**Comment by asomers** [comment]
Yes, it would be nice to have, gated by `nightly` of course.  Would you be willing to try implementing it?

**Comment by Enes1313** [comment]
I edited the top comment before your new message :).

First of all, I will try to do your first suggestion. If I can't, I will try implement the nightly feature.

**Comment by asomers** [comment]
PR #508 shows an example of using variadic functions.  However, the variadic arguments aren't currently available to the matcher and returner functions.  That feature could be added.  However, it probably won't be possible to use a `.with()` matcher.  Only `.withf()` matchers will be possible, due to lifetime problems, similar to how `#[concretize]` works.  I'm going to close this issue now, and defer this enhancement to the future.

Please open a new issue if anybody needs that feature.
